File: tuikit-atomic-x/ios/view/SVGAAnimationViewManager.m
#import <React/RCTViewManager.h>
#import <React/RCTUIManager.h>

@interface RCT_EXTERN_MODULE(SVGAAnimationViewManager, RCTViewManager)

RCT_EXPORT_VIEW_PROPERTY(onFinished, RCTDirectEventBlock)

RCT_EXTERN_METHOD(startAnimation:(nonnull NSNumber *)node url:(NSString *)url)
RCT_EXTERN_METHOD(stopAnimation:(nonnull NSNumber *)node)

@end

File: tuikit-atomic-x/ios/view/SVGAAnimationViewManager.swift
import Foundation
import UIKit
import React

@objc(SVGAAnimationViewManager)
class SVGAAnimationViewManager: RCTViewManager {
    
    override static func requiresMainQueueSetup() -> Bool {
        return true
    }
    
    override func view() -> UIView! {
        return SVGAAnimationView()
    }
    
    @objc
    func startAnimation(_ node: NSNumber, url: String) {
        DispatchQueue.main.async {
            guard let bridge = self.bridge else {
                print("SVGAAnimationViewManager: bridge is nil")
                return
            }
            guard let view = bridge.uiManager.view(forReactTag: node) as? SVGAAnimationView else {
                print("SVGAAnimationViewManager: startAnimation - view not found for node: \(node)")
                return
            }
            view.startAnimation(url)
        }
    }
    
    @objc
    func stopAnimation(_ node: NSNumber) {
        DispatchQueue.main.async {
            guard let bridge = self.bridge else {
                print("SVGAAnimationViewManager: bridge is nil")
                return
            }
            guard let view = bridge.uiManager.view(forReactTag: node) as? SVGAAnimationView else {
                print("SVGAAnimationViewManager: stopAnimation - view not found for node: \(node)")
                return
            }
            view.stopAnimation()
        }
    }
}
